tools/sutf.cpp

```diff
@@ -0,0 +1,313 @@
+#include "sutf.h"
+
+#include <vector>
+#include <string>
+#include <set>
+#include <filesystem>
+#include <iterator>
+#include <climits>
+
+CommandLine parse_and_validate_arguments(int argc, char* argv[]) {
+  CommandLine cmdline;
+  std::vector<std::string> arguments;
+
+  for (int i=1; i < argc; i++) {
+    std::string arg{argv[i]};
+    if ((arg == "--help") || (arg == "-h")) {
+      CommandLine::show_help();
+      return cmdline;
+    }
+    else if ((arg == "--list") || (arg == "-l")) {
+      CommandLine::show_formats();
+      return cmdline;
+    }
+    else {
+      arguments.push_back(std::move(arg));
+    }
+  }
+
+  bool has_from_arg = false;
+  bool has_to_arg = false;
+
+  for (size_t i=0; i < arguments.size();) {
+    const std::string& arg = arguments[i];
+
+    if (arg == "-f") {
+      const std::string& value = arguments.at(i + 1);
+      cmdline.from_encoding = value;
+      has_from_arg = true;
+      i += 2;
+    }
+    else if (arg == "-t") {
+      const std::string& value = arguments.at(i + 1);
+      cmdline.to_encoding = value;
+      has_to_arg = true;
+      i += 2;
+    }
+    else if (arg == "-o") {
+      const std::string& value = arguments.at(i + 1);
+      cmdline.output_file = value;
+      i += 2;
+    }
+    else {
+      if (! std::filesystem::exists(arg)) {
+        throw std::runtime_error("File " + arg + " does not exist.");
+      }
+      cmdline.input_files.insert(arg);
+      i++;
+    }
+  }
+
+  if (!has_from_arg || !has_to_arg) {
+    throw std::invalid_argument("Missing -f or -t argument(s).");
+  }
+
+  return cmdline;
+}
+
+void CommandLine::run() {
+  if (output_file.empty()) {
+    run_procedure(stdout);
+  } else {
+    SIMDUTF_DISABLE_DEPRECATED_WARNING 
```

Comment:
@NicolasJiaxin 

I have figured it out.
You have to do...
```
  SIMDUTF_PUSH_DISABLE_WARNINGS
  SIMDUTF_DISABLE_DEPRECATED_WARNING
 something here
  SIMDUTF_POP_DISABLE_WARNINGS
```

Could you fix these? I think it is my fault, I called `SIMDUTF_DISABLE_DEPRECATED_WARNING` without first calling `SIMDUTF_PUSH_DISABLE_WARNINGS` which caused Visual Studio to get angry.